docs/src/index.md

```diff
@@ -31,23 +31,25 @@ Also see [Implementing pullbacks](@ref) on how to implement back-propagation for
 We will try a few things with the following functions:
 
 ```jldoctest rosenbrock
-julia> rosenbrock(x, y) = (1.0 - x)^2 + 100.0 * (y - x^2)^2
-rosenbrock (generic function with 1 method)
+julia> rosenbrock(x, y) = (1.0 - x)^2 + 100.0 * (y - x^2)^2;
 
-julia> rosenbrock_inp(x) = (1.0 - x[1])^2 + 100.0 * (x[2] - x[1]^2)^2
-rosenbrock_inp (generic function with 1 method)
+julia> rosenbrock_inp(x) = (1.0 - x[1])^2 + 100.0 * (x[2] - x[1]^2)^2;
 ```
 
+where we note for future reference that the value of this function at `x=1.0`, `y=2.0` is `100.0`, and its derivative
+with respect to `x` at that point is `-400.0`, and its derivative with respect to `y` at that point is `200.0`. 
+
 ## Reverse mode
 
 The return value of reverse mode [`autodiff`](@ref) is a tuple that contains as a first value
-the derivative value of the active inputs and optionally the primal return value.
+the derivative value of the active inputs and optionally the _primal_ return value (i.e. the 
+value of the undifferentiated function).
```

Comment:
Consider not using "value" to mean so many things here?

> is a tuple that contains as a first *element* the *derivatives* of ..., and optionally the primal *value* (i.e. what the function returns).

Reply:
Maybe "optionally" also seems a bit odd to describe the output not the input. It's not that you may omit this. It's that ReverseWithPrimal tells it to.

Reply:
yeah we definitely don't need to say "derivative value" and can just say "the derivative of"

Reply:
and "the value of the undifferentiated function" -> "the result of the original function without differentiation"

Reply:
Also consider putting the ReverseWithPrimal case first, as without it,  `((-400.0, 200.0),)` seems like a puzzle to count the brackets & guess why.

Reply:
oh yeah totally fair, if you want to put that in this PR that would be fine with me!

Reply:
I can't make suggestions across deleted lines :/  so this is going to be messy...